src/Elm/Parser/Expression.elm

```diff
@@ -667,16 +672,115 @@ operation currentPrecedence leftExpression =
         |> Combine.oneOf
 
 
-operations : Dict Int (Node Expression -> Parser State (Node Expression))
-operations =
-    andThenOneOf
-        |> List.map Tuple.first
-        |> List.Extra.unique
-        |> (::) 0
-        |> (::) 95
-        |> List.foldl
-            (\precedence dict -> Dict.insert precedence (operation precedence) dict)
-            Dict.empty
+getOperation : Int -> Maybe (Node Expression -> Parser State (Node Expression))
+getOperation precedence =
+    case precedence of
+        0 ->
+            Just operation0
+
+        1 ->
+            Just operation1
+
+        2 ->
+            Just operation2
+
+        3 ->
+            Just operation3
+
+        4 ->
+            Just operation4
+
+        5 ->
+            Just operation5
+
+        6 ->
+            Just operation6
+
+        7 ->
+            Just operation7
+
+        8 ->
+            Just operation8
+
+        9 ->
+            Just operation9
+
+        90 ->
+            Just operation90
+
+        95 ->
+            Just operation95
+
+        100 ->
+            Just operation100
+
+        _ ->
+            Nothing
+
+
+operation0 : Node Expression -> Parser State (Node Expression)
+operation0 leftExpression =
+    operation 0 leftExpression
+
+
+operation1 : Node Expression -> Parser State (Node Expression)
+operation1 leftExpression =
+    operation 1 leftExpression
+
+
+operation2 : Node Expression -> Parser State (Node Expression)
+operation2 leftExpression =
+    operation 2 leftExpression
+
+
+operation3 : Node Expression -> Parser State (Node Expression)
+operation3 leftExpression =
+    operation 3 leftExpression
+
+
+operation4 : Node Expression -> Parser State (Node Expression)
+operation4 leftExpression =
+    operation 4 leftExpression
+
+
+operation5 : Node Expression -> Parser State (Node Expression)
+operation5 leftExpression =
+    operation 5 leftExpression
+
+
+operation6 : Node Expression -> Parser State (Node Expression)
+operation6 leftExpression =
+    operation 6 leftExpression
+
+
+operation7 : Node Expression -> Parser State (Node Expression)
+operation7 leftExpression =
+    operation 7 leftExpression
+
+
+operation8 : Node Expression -> Parser State (Node Expression)
+operation8 leftExpression =
+    operation 8 leftExpression
+
+
+operation9 : Node Expression -> Parser State (Node Expression)
+operation9 leftExpression =
+    operation 9 leftExpression
+
+
+operation90 : Node Expression -> Parser State (Node Expression)
+operation90 leftExpression =
+    operation 90 leftExpression
+
+
+operation95 : Node Expression -> Parser State (Node Expression)
+operation95 leftExpression =
+    operation 95 leftExpression
+
+
+operation100 : Node Expression -> Parser State (Node Expression)
+operation100 leftExpression =
+    operation 100 leftExpression
```

Comment:
How much faster is this? Because it seems a little excessive :)

Reply:
indeed, benchmarking rn. There are likely more low-hanging fruits (e.g. the List.map inside oneOf there that can be moved into the filterMap one level higher)

Reply:
benchmarks show predefining these doesn't make a difference, you're right.

Reply:
I think the way it's done right now, we're not actually precomputing anything, and instead recomputing (filter the precedences) every time. I'm slightly in doubt as to whether precomputation worked before this PR either, though I at some believed I had made it work.

If we got rid of the expression argument in `operator` (probably not possible?), or if we did the precedence filtering and then returned a lambda that took the expression argument, then it would work for sure.

Reply:
are you looking at the latest state where we have module-level filtered lists?
And I also don't think it worked in the `Dict` version because the filterMap is inside the constructed lambda for each entry. But I haven't tested so who knows

Reply:
Sorry, what do you mean with "module-level filtered lists"?
And yeah, I don't think it worked with `Dict` either, although at the time I did think so. I'll look into it at some point again.